src/solarFunctions.f90

```diff
@@ -109,7 +110,12 @@ subroutine calcZenith( t, dec )
     ! The local hour angle is the angle between the observer's
     ! meridian and the Sun's meridian. Time must be in GMT/UTC and
     ! longitude in degrees.
-    currentFracDay = currentDayOfYear + ( t / 86400.0_DP )
+    !
+    ! TODO: currentFracDay is wrong because t/86400 is not the
```

Comment:
So what's wrong here? The fact there aren't exactly 86400 seconds in a day?

Reply:
No I think that `t` is not the number of seconds. What happens is that `currentFracDay` increases too rapidly. For example:

currentDayOfYear    currentFracDay
1                                 1.8
1                                1.9
2                                2.1
2                                3.2

You notice that only if the model is run for more than one day.  It's really just fractional part we are interested in, so if there is an easier way to calculate `currentFracHour` then we can get rid of it (Hope it makes sense)